subprojects/plugins/src/main/java/org/gradle/api/internal/tasks/DefaultGroovySourceSet.java

```diff
@@ -36,14 +36,14 @@ public abstract class DefaultGroovySourceSet implements GroovySourceSet, HasPubl
     private final SourceDirectorySet allGroovy;
 
     @Inject
-    public DefaultGroovySourceSet(String name, String displayName, ObjectFactory objectFactory) {
-        this.groovy = createGroovySourceDirectorySet(name, displayName, objectFactory);
+    public DefaultGroovySourceSet(String name, String displayName, ObjectFactory objectFactory, TaskDependencyFactory taskDependencyFactory) {
+        this.groovy = createGroovySourceDirectorySet(name, displayName, objectFactory, taskDependencyFactory);
         allGroovy = objectFactory.sourceDirectorySet("all" + name, displayName + " Groovy source");
         allGroovy.source(groovy);
         allGroovy.getFilter().include("**/*.groovy");
     }
 
-    private static GroovySourceDirectorySet createGroovySourceDirectorySet(String name, String displayName, ObjectFactory objectFactory) {
+    private static GroovySourceDirectorySet createGroovySourceDirectorySet(String name, String displayName, ObjectFactory objectFactory, TaskDependencyFactory taskDependencyFactory) {
```

Comment:
❌ Remove unused parameter `taskDependencyFactory`.

Reply:
Fixed in https://github.com/gradle/gradle/pull/22433/commits/9f696b19a4ca50fbcc321eec029ec161c4885ac3